api/src/main/resources/api/service_openapi.yaml

```diff
@@ -75,6 +75,21 @@ paths:
       responses:
         200:
           $ref: '#/components/responses/GetEntityResponse'
+  '/underlays/{underlayName}/entities/{entityName}/filters':
+    parameters:
+      - $ref: '#/components/parameters/Underlay'
+      - $ref: '#/components/parameters/Entity'
+    get:
+      # TODO authorization
+      security: []
+      summary: |
+        Returns what attributes and relationships can be used to create an entity filter and hints
+        for their values.
+      operationId: getFilters
+      tags: [EntitiesFilters]
```

Comment:
I just now realized that `Entity` just has a `schema: string` field on it. Should this information here instead just be embedded onto the entity model itself? What layers of abstraction would this break?

Reply:
Sorry, I'm not sure what you mean. The `#/components/parameters/Entity` is the `{entityName}` parameter with the schema: string field. There's also a `#/components/schemas/Entity` that has `name` and `attributes` properties.
I don't know what information you're asking about.

Reply:
As in , this filter metadata. An alternative to this hierarchical API would be to simply return this filter information on the `Entity` model itself. Do you imagine cases where you'd want to grab the entity and its filter details separately?

Reply:
Ah, hmmm. I'm not sure if we'd ever want to grab them separately, or not care about one or the other.
Putting them on the model itself would start incorporating search concerns with the logical model. That seems fine-ish for how to filter entities/attributes, but I don't know if we'd keep doing that for other Tanagra searches. Like if we wanted to let the users choose what to group by on demographic count searches, would we also add a 'is_groupable' field to 'Attribute'?
I'm not sure to how to weigh some modeling purity vs a client use case. Is it annoying to do multiple backend calls? Are we going to want to just have a "tell me everything about this entity and its attributes" endpoint eventually anyway.
Don't know, happy to start in one direction then change later with client input as well.

Reply:
I think we have a lot of flexibility with what we decide in this case because these are static / read-only metadata endpoints. If we were to have full CRUD for these models, things would get a bit more complicated and I'd have to think harder (I might still suggest embedding on the entity resource, though). The client will almost certainly be fetching this all at once. 

My gut says that just encoding this onto the Entity model is probably the best path. I could be wrong about this though, and I'm fine with the way you have it, if you feel strongly on this point.

Reply:
SG, PTAL